antithesis test had 'please fix or report' message

it looks like we attempted to select beyond head potentially, haven't looked at trace fully.

Given this is onSlot, we can throw a better message and not fall through generic handling...
```
[   27564] [   897.401110] [service_teku-besu-0]                     [I] 2023-04-03 13:18:07.873 DEBUG - Ignoring calculated payload building attributes since it violates ordering
[   27564] [   897.403531] [service_teku-besu-0]                     [I] 2023-04-03 13:18:07.875 DEBUG - Ignoring calculated payload building attributes since it violates ordering
[   27564] [   897.413403] [service_prysm-geth-0]                    [I] time="2023-04-03 13:18:07" level=debug msg="gRPC request finished." backend=[] duration=167.668µs method="/ethereum.eth.v1alpha1.BeaconNodeValidator/GetSyncMessageBlockRoot"
[   27564] [   897.446149] [service_lighthouse-geth-0]               [I] : InsufficientPeers, service: libp2p
[   27564] [   897.446149] [service_lighthouse-geth-0]               [I] Apr 03 13:18:07.917 WARN Could not publish message               error: InsufficientPeers, service: libp2p
[   27564] [   897.446149] [service_lighthouse-geth-0]               [I] Apr 03 13:18:07.917 WARN Could not publish message               error: InsufficientPeers, service: libp2p
[   27564] [   897.446149] [service_lighthouse-geth-0]               [I] Apr 03 13:18:07.917 WARN Could not publish message               error: InsufficientPeers, service: libp2p
[   27564] [   897.525497] [service_nimbus-nethermind-0]             [I] DBG 2023-04-03 13:18:07.997+00:00 Current syncing state                      topics="syncman" workers_map=wwwwwUwwww sleeping_workers_count=0 waiting_workers_count=9 pending_workers_count=1 wall_head_slot=144 local_head_slot=103 pause_time= avg_sync_speed=0.2582259141418051 ins_sync_speed=0.0 sync_ident=main direction=forward
[   27564] [   897.528534] [service_lodestar-besu-0]                 [I] Apr-03 13:18:08.000[]                 [32minfo[39m: Searching peers - peers: 0 - slot: 144 (skipped 41) - head: 103 0xc19e…ef90 - execution: valid(0xcb17…264f) - finalized: 0x0000…0000:0
[   27564] [   897.531551] [service_lodestar-geth-0]                 [I] Apr-03 13:18:08.003[]                 [32minfo[39m: Searching peers - peers: 1 - slot: 144 (skipped 41) - head: 103 0xc19e…ef90 - execution: valid(0xcb17…264f) - finalized: 0x0000…0000:0
[   27564] [   897.533525] [service_teku-besu-0]                     [I] 2023-04-03 13:18:08.005 DEBUG - Producing instance of Job 'DEFAULT.Timer-1', class=tech.pegasys.teku.services.timer.ScheduledTimeEvent
[   27564] [   897.540343] [service_lighthouse-geth-0]               [I] Apr 03 13:18:08.006 INFO Connected to beacon node(s)             synced: 1, available: 1, total: 1, service: notifier
[   27564] [   897.540343] [service_lighthouse-geth-0]               [I] Apr 03 13:18:08.012 INFO Some validators active                  slot: 144, epoch: 18, total_validators: 10, active_validators: 7, current_epoch_proposers: 2, service: notifier
[   27564] [   897.541525] [service_lighthouse-nethermind-0]         [I] Apr 03 13:18:08.002Apr 03 13:18:08.003 INFO Connected to beacon node(s)             synced: 1, available: 1, total: 1, service: notifier
[   27564] [   897.541525] [service_lighthouse-nethermind-0]         [I] Apr 03 13:18:08.013 INFO Awaiting activation                     slot: 144, epoch: 18, validators: 10, service: notifier
[   27564] [   897.547076] [service_lighthouse-besu-0]               [I] Apr 03 13:18:08.019 INFO Connected to beacon node(s)             synced: 1, available: 1, total: 1, service: notifier
[   27564] [   897.547076] [service_lighthouse-besu-0]               [I] Apr 03 13:18:08.019 INFO Some validators active                  slot: 144, epoch: 18, total_validators: 10, active_validators: 9, current_epoch_proposers: 0, service: notifier
[   27564] [   897.550122] [service_lodestar-nethermind-0]           [I] Apr-03 13:18:08.020[]                 [32minfo[39m: Searching peers - peers: 1 - slot: 144 (skipped 41) - head: 103 0xc19e…ef90 - execution: valid(0xcb17…264f) - finalized: 0x0000…0000:0
[   27564] [   897.550569] [service_teku-whale-0]                    [I] 2023-04-03 13:18:08.021 DEBUG - Producing instance of Job 'DEFAULT.Timer-1', class=tech.pegasys.teku.services.timer.ScheduledTimeEvent
[   27564] [   897.550569] [service_teku-whale-0]                    [I] 2023-04-03 13:18:08.022 DEBUG - Calling execute on job DEFAULT.Timer-1
[   27564] [   897.552701] [service_teku-besu-0]                     [I] 2023-04-03 13:18:08.024 DEBUG - Calling execute on job DEFAULT.Timer-1
[   27564] [   897.561474] [service_teku-whale-0]                    [I] 2023-04-03 13:18:08.033 DEBUG - batch acquisition of 1 triggers
[   27564] [   897.578620] [service_teku-besu-0]                     [I] 2023-04-03 13:18:08.050 INFO  - Validator   *** Published attestation        Count: 2, Slot: 144, Root: b60d23b6ce81a34f1b2f5da5738aa1c9c99053936028063117a4a757a3641819
[   27564] [   897.594806] [service_lighthouse-besu-0]               [I] Apr 03 13:18:08.037 WARN Low peer count                          peer_count: 0, service: slot_notifier
[   27564] [   897.654597] [service_teku-nethermind-0]               [I] 2023-04-03 13:18:08.1246|Waiting for peers... 37s
[   27564] [   897.666472] [service_teku-besu-0]                     [I] 2023-04-03 13:18:08.138 DEBUG - batch acquisition of 1 triggers
[   27564] [   897.667627] [service_nimbus-geth-0]                   [I] DBG 2023-04-03 13:18:08.139+00:00 Timed out or unrequested whoareyou packet  topics="eth p2p discv5" address=10.0.20.70:4000
[   27564] [   897.669666] [service_lighthouse-nethermind-0]         [I] WARN Low peer count                          peer_count: 0, service: slot_notifier
[   27564] [   897.680460] [service_teku-besu-0]                     [I] 2023-04-03 13:18:08.152 INFO  - Validator   *** Published sync_signature     Count: 3, Slot: 144, Root: b60d23b6ce81a34f1b2f5da5738aa1c9c99053936028063117a4a757a3641819
[   27564] [   897.690612] [service_lighthouse-besu-0]               [I] Apr 03 13:18:08.103 INFO Synced                                  slot: 144, block:    …  empty, epoch: 18, finalized_epoch: 0, finalized_root: 0x69a9…f4ab, exec_hash: 0x05ec…eb48 (verified), peers: 0, service: slot_notifier
[   27564] [   897.690612] [service_lighthouse-besu-0]               [I] Apr 03 13:18:08.162 WARN Syncing deposit contract block cache    est_blocks_remaining: 6, service: slot_notifier
[   27564] [   897.708083] [service_nimbus-nethermind-0]             [I] DBG 2023-04-03 13:18:08.179+00:00 Timed out or unrequested whoareyou packet  topics="eth p2p discv5" address=10.0.20.70:4000
[   27564] [   897.711650] [service_prysm-geth-0]                    [I] time="2023-04-03 13:18:07" level=error msg="Could not request sync message block root to sign" error="rpc error: code = DeadlineExceeded desc = context deadline exceeded" prefix=validator
[   27564] [   897.770081] [service_teku-geth-0]                     [I] 2023-04-03 13:18:08.087 ERROR - PLEASE FIX OR REPORT | Unexpected exception thrown for event 'interface tech.pegasys.teku.ethereum.events.SlotEventsChannel.onSlot' in handler 'tech.pegasys.teku.validator.coordinator.performance.DefaultPerformanceTracker'
[   27564] [   897.770081] [service_teku-geth-0]                     [I] java.util.concurrent.CompletionException: java.lang.IllegalArgumentException: Block at slot 133 not available from state at slot 132
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315) ~[?:?]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320) ~[?:?]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at java.util.concurrent.CompletableFuture.biApply(CompletableFuture.java:1313) ~[?:?]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at java.util.concurrent.CompletableFuture.biApplyStage(CompletableFuture.java:1329) ~[?:?]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at java.util.concurrent.CompletableFuture.thenCombine(CompletableFuture.java:2210) ~[?:?]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at tech.pegasys.teku.infrastructure.async.SafeFuture.thenCombine(SafeFuture.java:536) ~[teku-infrastructure-async-23.3.1.jar:23.3.1]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at tech.pegasys.teku.validator.coordinator.performance.DefaultPerformanceTracker.getAttestationPerformanceForEpoch(DefaultPerformanceTracker.java:264) ~[teku-beacon-validator-23.3.1.jar:23.3.1]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at tech.pegasys.teku.validator.coordinator.performance.DefaultPerformanceTracker.reportAttestationPerformance(DefaultPerformanceTracker.java:181) ~[teku-beacon-validator-23.3.1.jar:23.3.1]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at tech.pegasys.teku.validator.coordinator.performance.DefaultPerformanceTracker.onSlot(DefaultPerformanceTracker.java:126) ~[teku-beacon-validator-23.3.1.jar:23.3.1]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at jdk.internal.reflect.GeneratedMethodAccessor4.invoke(Unknown Source) ~[?:?]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at tech.pegasys.teku.infrastructure.events.DirectEventDeliverer.executeMethod(DirectEventDeliverer.java:74) ~[teku-infrastructure-events-23.3.1.jar:23.3.1]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at tech.pegasys.teku.infrastructure.events.DirectEventDeliverer.deliverTo(DirectEventDeliverer.java:54) ~[teku-infrastructure-events-23.3.1.jar:23.3.1]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at tech.pegasys.teku.infrastructure.events.AsyncEventDeliverer.lambda$deliverTo$0(AsyncEventDeliverer.java:66) ~[teku-infrastructure-events-23.3.1.jar:23.3.1]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at tech.pegasys.teku.infrastructure.events.AsyncEventDeliverer$QueueReader.deliverNextEvent(AsyncEventDeliverer.java:125) [teku-infrastructure-events-23.3.1.jar:23.3.1]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at tech.pegasys.teku.infrastructure.events.AsyncEventDeliverer$QueueReader.run(AsyncEventDeliverer.java:116) [teku-infrastructure-events-23.3.1.jar:23.3.1]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at java.lang.Thread.run(Thread.java:833) [?:?]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] Caused by: java.lang.IllegalArgumentException: Block at slot 133 not available from state at slot 132
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at com.google.common.base.Preconditions.checkArgument(Preconditions.java:435) ~[guava-31.1-jre.jar:?]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at tech.pegasys.teku.spec.logic.common.helpers.BeaconStateAccessors.getBlockRootAtSlot(BeaconStateAccessors.java:235) ~[teku-ethereum-spec-23.3.1.jar:23.3.1]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at tech.pegasys.teku.spec.Spec.getBlockRootAtSlot(Spec.java:423) ~[teku-ethereum-spec-23.3.1.jar:23.3.1]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at tech.pegasys.teku.validator.coordinator.performance.DefaultPerformanceTracker.calculateAttestationPerformance(DefaultPerformanceTracker.java:317) ~[teku-beacon-validator-23.3.1.jar:23.3.1]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at tech.pegasys.teku.validator.coordinator.performance.DefaultPerformanceTracker.lambda$getAttestationPerformanceForEpoch$7(DefaultPerformanceTracker.java:267) ~[teku-beacon-validator-23.3.1.jar:23.3.1]
[   27564] [   897.770081] [service_teku-geth-0]                     [I] at java.util.concurrent.CompletableFuture.biApply(CompletableFuture.java:1311) ~[?:?]
```

--- a/beacon/validator/build.gradle
+++ b/beacon/validator/build.gradle
@@ -37,6 +37,7 @@ dependencies {
   testImplementation testFixtures(project(':ethereum:networks'))
   testImplementation testFixtures(project(':storage'))
   testImplementation testFixtures(project(':infrastructure:async'))
+  testImplementation testFixtures(project(':infrastructure:logging'))
   testImplementation testFixtures(project(':infrastructure:metrics'))
   testImplementation testFixtures(project(':infrastructure:time'))
 

--- a/beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/performance/DefaultPerformanceTracker.java
+++ b/beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/performance/DefaultPerformanceTracker.java
@@ -36,6 +36,8 @@ import java.util.concurrent.ConcurrentSkipListMap;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.apache.tuweni.bytes.Bytes32;
 import tech.pegasys.teku.infrastructure.async.SafeFuture;
 import tech.pegasys.teku.infrastructure.logging.StatusLogger;
@@ -53,6 +55,7 @@ import tech.pegasys.teku.validator.api.ValidatorPerformanceTrackingMode;
 import tech.pegasys.teku.validator.coordinator.ActiveValidatorTracker;
 
 public class DefaultPerformanceTracker implements PerformanceTracker {
+  private static final Logger LOG = LogManager.getLogger();
 
   @VisibleForTesting
   final NavigableMap<UInt64, Set<SlotAndBlockRoot>> producedBlocksByEpoch =
@@ -136,7 +139,7 @@ public class DefaultPerformanceTracker implements PerformanceTracker {
       reportingTasks.add(reportSyncCommitteePerformance(currentEpoch));
     }
 
-    SafeFuture.allOf(reportingTasks.toArray(SafeFuture[]::new)).join();
+    SafeFuture.allOf(reportingTasks.toArray(SafeFuture[]::new)).handleException(LOG::error).join();
   }
 
   private SafeFuture<?> reportBlockPerformance(final UInt64 currentEpoch) {

--- a/beacon/validator/src/test/java/tech/pegasys/teku/validator/coordinator/performance/DefaultPerformanceTrackerTest.java
+++ b/beacon/validator/src/test/java/tech/pegasys/teku/validator/coordinator/performance/DefaultPerformanceTrackerTest.java
@@ -19,12 +19,14 @@ import static org.mockito.ArgumentMatchers.anyString;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
 import static org.mockito.Mockito.when;
 import static tech.pegasys.teku.validator.coordinator.performance.DefaultPerformanceTracker.ATTESTATION_INCLUSION_RANGE;
 
 import java.util.List;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import tech.pegasys.infrastructure.logging.LogCaptor;
 import tech.pegasys.teku.bls.BLSKeyGenerator;
 import tech.pegasys.teku.bls.BLSKeyPair;
 import tech.pegasys.teku.bls.BLSTestUtil;
@@ -360,6 +362,41 @@ public class DefaultPerformanceTrackerTest {
     performanceTracker.onSlot(spec.computeStartSlotAtEpoch(epoch));
     verify(log).performance(performance.toString());
     verify(validatorPerformanceMetrics).updateSyncCommitteePerformance(performance);
+  }
+
+  @Test
+  void shouldHandleErrorsWhenReportTasksFail() {
+    chainUpdater.updateBestBlock(chainUpdater.advanceChainUntil(1));
+    final Attestation attestation = createAttestationForParentBlockOnSlot(1);
+    final UInt64 slot = spec.computeStartSlotAtEpoch(ATTESTATION_INCLUSION_RANGE);
+
+    performanceTracker.saveProducedAttestation(attestation);
+    when(validatorTracker.getNumberOfValidatorsForEpoch(any())).thenThrow(new RuntimeException());
+
+    try (LogCaptor logCaptor = LogCaptor.forClass(DefaultPerformanceTracker.class)) {
+      performanceTracker.onSlot(slot);
+
+      // No attestation performance report on status logger because task failed
+      verifyNoInteractions(log);
+      assertThat(logCaptor.getErrorLogs()).hasSize(1);
+    }
+  }
+
+  /**
+   * Creates an attestation voting for block on the slot provided. The attestation will be included
+   * in block slot + 1.
+   *
+   * @param slot the slot of the block being attested
+   * @return the created attestation
+   */
+  private Attestation createAttestationForParentBlockOnSlot(int slot) {
+    Attestation attestationForBlock1 = createAttestation(slot + 1, slot);
+    ChainBuilder.BlockOptions block2Options = ChainBuilder.BlockOptions.create();
+    block2Options.addAttestation(attestationForBlock1);
+    SignedBlockAndState latestBlockAndState = chainBuilder.generateBlockAtSlot(2, block2Options);
+    chainUpdater.saveBlock(latestBlockAndState);
+    chainUpdater.updateBestBlock(latestBlockAndState);
+    return attestationForBlock1;
   }
 
   private Attestation createAttestation(

